File: stepik-union/src/main/java/org/stepik/core/StudyBasePluginConfigurator.java
package org.stepik.core;

import com.intellij.openapi.actionSystem.DefaultActionGroup;
import com.intellij.openapi.fileEditor.FileEditorManagerAdapter;
import com.intellij.openapi.fileEditor.FileEditorManagerEvent;
import com.intellij.openapi.fileEditor.FileEditorManagerListener;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.vfs.VirtualFile;
import org.stepik.core.courseFormat.StudyNode;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;
import java.util.Collections;
import java.util.Map;

public abstract class StudyBasePluginConfigurator implements StudyPluginConfigurator {
    @NotNull
    public static DefaultActionGroup getDefaultActionGroup() {
        return new DefaultActionGroup();
    }

    @NotNull
    @Override
    public DefaultActionGroup getActionGroup(Project project) {
        return getDefaultActionGroup();
    }

    @NotNull
    @Override
    public Map<String, JPanel> getAdditionalPanels(Project project) {
        return Collections.emptyMap();
    }

    @NotNull
    @Override
    public FileEditorManagerListener getFileEditorManagerListener(@NotNull Project project) {
        return new FileEditorManagerAdapter() {
            @Override
            public void selectionChanged(@NotNull FileEditorManagerEvent event) {
                VirtualFile file = event.getNewFile();
                if (file == null) {
                    return;
                }

                StudyNode stepNode = StudyUtils.getStudyNode(project, file);

                if (stepNode != null) {
                    StepikProjectManager.setSelected(project, stepNode);
                }
            }
        };
    }
}

File: stepik-core/src/main/kotlin/org/stepik/core/StudyBasePluginConfigurator.kt
package org.stepik.core

import com.intellij.openapi.actionSystem.DefaultActionGroup
import com.intellij.openapi.fileEditor.FileEditorManagerEvent
import com.intellij.openapi.fileEditor.FileEditorManagerListener
import com.intellij.openapi.project.Project
import org.stepik.core.StudyUtils.getProjectManager
import org.stepik.core.StudyUtils.getStudyNode
import javax.swing.JPanel

abstract class StudyBasePluginConfigurator : StudyPluginConfigurator {

    override fun getActionGroup(project: Project): DefaultActionGroup {
        return defaultActionGroup
    }

    override fun getAdditionalPanels(project: Project): Map<String, JPanel> {
        return emptyMap()
    }

    override fun getFileEditorManagerListener(project: Project): FileEditorManagerListener {
        return object : FileEditorManagerListener {
            override fun selectionChanged(event: FileEditorManagerEvent) {
                val file = event.newFile ?: return

                getProjectManager(project)?.selected = getStudyNode(project, file) ?: return
            }
        }
    }

    companion object {
        val defaultActionGroup: DefaultActionGroup
            get() = DefaultActionGroup()
    }
}
